tests/GoodsDetail.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+
+import { GoodsDetail } from "@/components/features/Goods/Detail";
+import { OptionSection } from "@/components/features/Goods/Detail/OptionSection";
+
+const GoodsData = {
+  id: 3245119,
+  name: '[단독각인] 피렌체 1221 에디션 오드코롱 50ml (13종 택1)',
+  imageUrl: 'https://st.kakaocdn.net/product/gift/product/20240215083306_8e1db057580145829542463a84971ae3.png',
+  price: 145000,
+};
+
+const queryClient = new QueryClient();
+
+test('상품의 상세 정보를 올바르게 렌더링', async () => {
+  render(
+    <QueryClientProvider client={queryClient}>
+      <GoodsDetail productId={GoodsData.id.toString()} />
+    </QueryClientProvider>
+  );
+
+  const name = await screen.findByText(GoodsData.name);
+  const image = screen.getByAltText(GoodsData.name);
+  const price = screen.getByText(`${GoodsData.price}원`);
+
+  await waitFor(() => {
+    expect(name).toBeInTheDocument();
+    expect(image).toBeInTheDocument();
+    expect(price).toBeInTheDocument();
+  });
```

Comment:
```suggestion
  expect(name).toBeInTheDocument();
  expect(image).toBeInTheDocument();
  expect(price).toBeInTheDocument();
```

waitFor가 없어도 테스트 실행할거라 불필요한 await 같아요. 이런건 테스트 실행시간을 늦춰주는 원인이 되기에 정리해주는게 좋습니다.